test/fork/stagedProposalProcessor.t.sol

```diff
@@ -0,0 +1,384 @@
+
+        // advance to stage 2
+        _test_advanceToStageX(2);
+
+        // execute proposal
+        _test_advanceLastStage();
+    }
+
+    function test_proposalFlowExecutingLastStage() public {
+        // create proposal
+        _test_createProposal();
+
+        // advance to stage 1
+        _test_advanceToStageX(1);
+
+        // advance to stage 2
+        _test_advanceToStageX(2);
+
+        // execute proposal
+        _test_executeLastStage();
+    }
+
+    function _test_createProposal() public {
+        uint64 startDate = uint64(block.timestamp);
+        bytes memory testMetadata = "test proposal metadata";
+        Action[] memory actions = new Action[](1);
+        actions[0].to = address(target);
+        actions[0].value = 0;
+        actions[0].data = abi.encodeCall(target.setValue, TARGET_VALUE);
+
+        // check proposal creation event was emitted
+
+        vm.expectEmit({
+            checkTopic1: false,
+            checkTopic2: true,
+            checkTopic3: true,
+            checkData: true,
+            emitter: address(sppPlugin)
+        });
+        emit ProposalCreated({
+            proposalId: 0,
+            creator: deployer,
+            startDate: startDate,
+            endDate: 0,
+            metadata: testMetadata,
+            actions: actions,
+            allowFailureMap: 0
+        });
+
+        // check sub proposal creation event was emitted
+
+        // todo check why this log is failing
+        // vm.expectEmit({
+        //     checkTopic1: false,
+        //     checkTopic2: false,
+        //     checkTopic3: false,
+        //     checkData: false,
+        //     emitter: address(sppPlugin)
+        // });
+
+        // emit SPP.SubProposalCreated({
+        //     proposalId: 0,
+        //     stageId: 0,
+        //     body: adminPlugin,
+        //     bodyProposalId: 0
+        // });
+
+        proposalId = sppPlugin.createProposal({
+            _metadata: testMetadata,
+            _actions: actions,
+            _allowFailureMap: 0,
+            _startDate: startDate,
+            _proposalParams: new bytes[][](0)
+        });
+
+        // check proposal
+        SPP.Proposal memory proposal = sppPlugin.getProposal(proposalId);
+
+        assertEq(
+            proposal,
+            SPP.Proposal({
+                allowFailureMap: 0,
+                lastStageTransition: startDate,
+                actions: actions,
+                stageConfigIndex: 1,
+                currentStage: 0,
+                executed: false,
+                canceled: false,
+                targetConfig: IPlugin.TargetConfig({
+                    target: address(dao),
+                    operation: IPlugin.Operation.Call
+                }),
+                creator: deployer
+            }),
+            "proposal"
+        );
+
+        // check subproposal id was stored
+        assertNotEq(sppPlugin.getBodyProposalId(proposalId, 0, adminPlugin), 0, "subproposalId");
```

Comment:
also, check that is not equal to `PROPOSAL_WITHOUT_ID`

Reply:
done https://github.com/aragon/staged-proposal-processor-plugin/pull/50/commits/c32997671601ed3a18e843060e0e02a7ab6dd746